src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+            {
+                if (TryGetFunctionDescriptor(node, null, dsFilePath, classNode.ClassName, customizationFile, out ReflectionFunctionDescriptor des))
+                {
+                    descriptors.Add(des);
+                }
+            }
+            return descriptors;
+        }
+
+        private static void AddZeroTouchNodesToSearch(IEnumerable<FunctionGroup> functionGroups, ref NodeSearchModel nodeSearchModel)
+        {
+            foreach (var funcGroup in functionGroups)
+            {
+                AddZeroTouchNodeToSearch(funcGroup, ref nodeSearchModel);
+            }
+        }
+
+        private static void AddZeroTouchNodeToSearch(FunctionGroup funcGroup, ref NodeSearchModel nodeSearchModel)
+        {
+            foreach (var functionDescriptor in funcGroup.Functions)
+            {
+                if (functionDescriptor.IsVisibleInLibrary)
+                {
+                    nodeSearchModel.Add(new ReflectionZeroTouchSearchElement(functionDescriptor as ReflectionFunctionDescriptor));
+                }
+            }
+        }
+
+        private static void FileInfosFromNodeModels(Assembly asm, Type nodeModelType, ref NodeSearchModel searchModel)
+        {
+
+            var fileInfos = new List<MdFileInfo>();
+            System.Type[] typesInAsm = null;
+            try
+            {
+                typesInAsm = asm.GetTypes();
+            }
+            catch (ReflectionTypeLoadException ex)
+            {
+                typesInAsm = ex.Types;
+            }
+
+            var nodeTypes = typesInAsm
+                .Where(x => NodeModelAssemblyLoader.IsNodeSubTypeReflectionLoaded(x, nodeModelType))
+                .Select(t => new TypeLoadData(t, t.GetAttributesFromReflectionContext()))
+                .Where(type => type != null && !type.IsDeprecated && !type.IsHidden)
+                .ToList();
+
+            foreach (var type in nodeTypes)
+            {
+                searchModel.Add(new NodeModelSearchElement(type, false));
+            }
+        }
+
+        private static List<string> GetDefaultPaths()
+        {
+            string[] runtimeAssemblies = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll", SearchOption.AllDirectories);
+            var dynamoDlls = new DirectoryInfo(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)).EnumerateFiles("*.dll").Select(x => x.FullName);
+
+            var paths = new List<string>(runtimeAssemblies);
+            paths.AddRange(dynamoDlls);
+
+            return paths;
+        }
+
+        private static bool TryGetFunctionDescriptor(
+            AssociativeNode associativeNode, Assembly asm, string asmPath,
+            string className, LibraryCustomization customization,
+            out ReflectionFunctionDescriptor descriptor)
+        {
+            descriptor = null;
+            string name = associativeNode.Name;
+            if (CoreUtils.IsSetter(name) ||
```

Comment:
why are setters skipped? What about getters?

Reply:
As far as i could see only setters should be skipped
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Library/LibraryServices.cs#L955

Getters are used as nodes
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCore/Library/LibraryServices.cs#L1003